ApplicationLibCode/Commands/ExportCommands/RicCreateDepthAdjustedLasFilesImpl.cpp

```diff
@@ -0,0 +1,403 @@
+                break;
+            }
+        }
+    }
+    return lasWellDepthValueAndIndexPerKLayer;
+}
+
+//--------------------------------------------------------------------------------------------------
+/// NOTE: map createIndexKDepthDataMapFromCase is created using well extractor, while sourceWellLogData depth
+/// values are from LAS file. Floating point rounding in LAS file can occur, thus depth values might be placed
+/// outside of K-layer close to top/bottom due to inaccuracy.
+//--------------------------------------------------------------------------------------------------
+void RicCreateDepthAdjustedLasFilesImpl::createDestinationWellsLasFiles( RimCase*                        selectedCase,
+                                                                         RimWellPath*                    sourceWell,
+                                                                         const std::vector<RimWellPath*> destinationWells,
+                                                                         const std::vector<QString>& selectedResultProperties,
+                                                                         const QString&              exportFolder,
+                                                                         double                      rkbDiff )
+{
+    auto*      sourceWellLogData  = sourceWell->wellLogFiles()[0]->wellLogFileData();
+    const auto defaultPropertyMap = createDefaultPropertyMap( selectedResultProperties, sourceWellLogData );
+
+    // NOTE: map createIndexKDepthDataMapFromCase is created using well extractor, while sourceWellLogData depth
+    // values are from LAS file. Floating point rounding in LAS file can occur, thus depth values might be placed
+    // outside of K-layer close to top/bottom due to inaccuracy.
+    const auto sourceWellDepthIndexAndPercValuePerKLayer =
+        createLasDepthIndexAndPercValuePerKLayerFromMap( sourceWellLogData->depthValues(),
+                                                         createIndexKDepthDataMapFromCase( selectedCase, sourceWell ) );
+    for ( RimWellPath* well : destinationWells )
+    {
+        const std::map<int, IndexKDepthData> destinationWellIndexKDepthsMap =
+            createIndexKDepthDataMapFromCase( selectedCase, well );
+
+        if ( destinationWellIndexKDepthsMap.empty() ) continue;
+
+        std::vector<double>                    mdValues;
+        std::vector<double>                    tvdMslValues;
+        std::vector<double>                    tvdRkbValues;
+        std::map<QString, std::vector<double>> propertyMap = defaultPropertyMap;
+        for ( const auto& [indexK, depthData] : destinationWellIndexKDepthsMap )
+        {
+            if ( !sourceWellDepthIndexAndPercValuePerKLayer.hasKLayer( indexK ) ) continue;
+
+            // Note destinationWellIndexKDepthsMap is retrieved from well extractors, and
```

Comment:
Incomplete comment

Reply:
Removed as comment earlier in function is describing the issue